pkg/controller/storagecluster/uninstall_reconciler_test.go

```diff
@@ -102,3 +104,53 @@ func assertCephClusterCleanupPolicy(
 	assert.Equal(t, CleanupPolicyConfirmation, cephCluster.Spec.CleanupPolicy.Confirmation)
 	assert.Equal(t, AllowUninstallWithVolumes, cephCluster.Spec.CleanupPolicy.AllowUninstallWithVolumes)
 }
+
+func TestDeleteStorageClasses(t *testing.T) {
+
+	testList := []struct {
+		isStorageClassExistBeforedelete bool
+		isStorageClassExistAfterdelete  bool
+	}{
+		// case 1: verify storage classes are present and delete them
+		{true, false},
+		// case 2: verify storage classes does not exist and delete should not get error out
+		{false, false},
+	}
```

Comment:
This is the right general idea, but it is too condensed to be immediately readable. I suggest something like this:

```suggestion
	cases := []struct {
		label string
		storageClassExists bool
	}{
		{
			label: "case 1", // verify storage classes are present and delete them
			storageClassExists: true,
		},
		{
			label: "case 2", // verify storage classes does not exist and delete should not get error out
			storageClassExists: false,
		},
	}
```